src/metabase/api/collection.clj

```diff
@@ -331,9 +334,10 @@
     ;; don't get models back from ulterior over-query
     ;; Previous examination with logging to DB says that there's no N+1 query for this.
     ;; However, this was only tested on H2 and Postgres
-    (assoc (dissoc row :collection_position :display :moderated_status :icon)
-           :can_write
-           (mi/can-write? Collection (:id row)))))
+    (cond-> row
+      (:personal_owner_id row) (assoc :name (collection/user->personal-collection-name (:personal_owner_id row) :user))
```

Comment:
I was a little confused at first how personal collections could ever be the children of other collections. Then I realized that that can be the case when fetching the root collection. Maybe it would be worth leaving a brief comment here mentioning this case?